src/features/ValueEditor/ValueEditor.tsx

```diff
@@ -22,10 +21,69 @@ import {
   FiPlus,
 } from 'react-icons/fi'
 import {LuEraser} from 'react-icons/lu'
-import {Validation} from '../../types'
+import {ZodSchema, z} from 'zod'
+import {generateErrorMessage} from 'zod-error'
 import {CopyButton} from '../CopyButton'
 import classes from './ValueEditor.module.css'
 
+const evaluateExpression = (expression: string) => {
+  try {
+    const evaluatedExpression = new Function(`return ${expression}`)()
+    return {success: true, data: evaluatedExpression} as const
+  } catch (e) {
+    return {
+      success: false,
+      error:
+        e instanceof SyntaxError
+          ? 'Invalid syntax'
+          : e instanceof ReferenceError
+            ? 'Invalid reference'
+            : 'Invalid value',
+    } as const
+  }
+}
+
+type ValidationResult =
+  | {
+      success: false
+      error: string
+    }
+  | {
+      success: true
+      parsedData: any
+    }
+
+const validateValue = (
+  schema: ZodSchema<any, z.ZodTypeDef, any> | undefined,
+  value: string,
+): ValidationResult => {
+  if (!schema) {
+    return {
+      success: false,
+      error: 'Invalid Schema',
+    }
+  }
+
+  const evaluatedValue = evaluateExpression(value)
+  if (!evaluatedValue.success) return evaluatedValue
+
+  try {
+    const validationRes = schema.safeParse(evaluatedValue.data)
+
+    return validationRes.success
+      ? ({success: true, parsedData: validationRes.data} as const)
+      : ({
+          success: false,
+          error: generateErrorMessage(validationRes.error.issues),
+        } as const)
+  } catch (e) {
```

Comment:
In general I really dislike that parenthesis style, with the catch on a newline my eye immediately recognize it as another block code. 
Imho it sucks expecially in cases like this when there's a return and ends up vertically aligned with the catch.
But consistency wins, either change all catch blocks or keep them like this if you prefer to.